### DISCUSSION TITLE
Scan all assemblies for IJob's

### DISCUSSION THREAD

**Discussion by projectje** [quartznet/quartznet#1429] [Q&A]
I have 50 projects/assemblies and some of them contain IJobs , I thought I'd try to generalize this:

In my static extension method used in .netcore composition i added in `services.AddQuartz(q=>`

```
foreach (System.Reflection.Assembly? currentAsm in AppDomain.CurrentDomain.GetAssemblies())
                foreach (Type? currentType in currentAsm.GetTypes())
                    if (typeof(IJob).IsAssignableFrom(currentType) && currentType.IsClass && !currentType.IsAbstract)
                               q.AddJob<currentType>(configuration,log)
 
```
where the method trying to call is ` AddJob<T> `

But <T> does not accept currentType because it says "is a variable but used like a type" but `log.Debug("{type}", currentType.GetType().FullName);` gives me the correct name.
This is probably a generics question.

**Comment by lahma** [comment]
I think you should just call `q.AddJob(currentType);` ?
